pkg/utils/utils_virtual.go

```diff
@@ -205,6 +227,33 @@ func getOpenstackDataFromMetadataService() (metaData *OSPMetaData, networkData *
 	return metaData, networkData, nil
 }
 
+// getPCIAddressFromMACAddress returns the PCI address of a device given its MAC address
+func getPCIAddressFromMACAddress(macAddress string) (string, error) {
+	nics, err := ghw.Network()
```

Comment:
this will keep reading all NICs information from the filesystem/ethtool for every MAC address

Reply:
i dont follow, this is plain inefficient from software perspective.
while  query from the  system the same information over and over again while expecting to get the same result.

Reply:
1. How do you know upfront it's a performance issue in the system as a whole. I understand that it feels inefficient, and it probably is. However, does it mean we need to fix it now before it has been identified as a real issue?
2. A solution is a caching layer that memoizes calls to this functions (i.e. store a map of macAddress to PCI address in memory). While this is not complex in and of itself, it does mean more code that we need to maintain.

Maybe @adrianchiris has a different idea than 2 in mind? 2 will only help if we actually call this function multiple times frequently enough for it to matter.

Reply:
GetOpenstackData is called once via writer when daemon starts. so im not aiming at caching it.

what id like to see is essentially moving this line to the caller so that listing the nic happens only once, then the list is passed as parameter to this function.

when there  are multiple interfaces passed  to the VM (multiple NICs) this becomes quatratic
(instead of listing e.g 16 nics  once you would list 16 nics 16  times)

each time triggering OS operations (ethtool / sysfs).

so my idea is essentially a 2 line change for which we spent multiple lines discussing it :)